android/app/src/main/java/org/fossasia/openevent/adapters/GlobalSearchAdapter.java

```diff
@@ -0,0 +1,395 @@
+                break;
+            default:
+                break;
+        }
+        return resultHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+
+        switch (holder.getItemViewType()){
+            case TRACK:
+                TrackViewHolder trackSearchHolder = (TrackViewHolder)holder;
+                configureTrackHolder(trackSearchHolder,position);
+                break;
+            case SESSION:
+                SessionViewHolder sessionSearchHolder = (SessionViewHolder)holder;
+                configureSessionHolder(sessionSearchHolder,position);
+                break;
+            case SPEAKER:
+                SpeakerViewHolder speakerSearchHolder = (SpeakerViewHolder)holder;
+                configureSpeakerHolder(speakerSearchHolder,position);
+                break;
+            case LOCATION:
+                LocationViewHolder locationSearchHolder = (LocationViewHolder)holder;
+                configureLocationHolder(locationSearchHolder,position);
+                break;
+            case DIVIDER:
+                SearchResultTypeViewHolder resultTypeViewHolder = (SearchResultTypeViewHolder)holder;
+                configureHeaderHolder(resultTypeViewHolder,position);
+                break;
+            default:
+                Log.e("Error in Search", "Check onBindViewHolder results");
+                break;
+        }
+    }
+
+    public void configureTrackHolder(TrackViewHolder holder,int position){
+
+        final Track currentTrack = (Track)getItem(position);
+
+        holder.trackTitle.setText(currentTrack.getName());
+        if(!TextUtils.isEmpty(currentTrack.getDescription())) {
+            holder.trackDescription.setText(currentTrack.getDescription());
+        } else {
+            holder.trackDescription.setVisibility(View.GONE);
+        }
+
+        int trackColor = Color.parseColor(currentTrack.getColor());
+        TextDrawable drawable = drawableBuilder.build(String.valueOf(currentTrack.getName().charAt(0)), trackColor);
+        holder.trackImageIcon.setImageDrawable(drawable);
+        holder.trackImageIcon.setBackgroundColor(Color.TRANSPARENT);
+
+        // Store the track color in color cache
+        TrackColors.storeColor(currentTrack.getId(), trackColor);
+
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(context, TrackSessionsActivity.class);
+                intent.putExtra(ConstantStrings.TRACK, currentTrack.getName());
+
+                // Send Track ID to Activity to leverage color cache
+                intent.putExtra(ConstantStrings.TRACK_ID, currentTrack.getId());
+                context.startActivity(intent);
+            }
+        });
+
+    }
+
+    public void configureSessionHolder(SessionViewHolder holder,int position){
```

Comment:
Rather than recreating viewholders, you can add the login of binding view to data in the original viewholders and make them public

Then reuse them here

Read this
https://jonfhancock.com/your-viewholders-are-dumb-make-em-not-dumb-82e6f73f630c?gi=f26b27eb8c17